Migrate storybook entries of QuantitySelector component

sunyatasattva opened the pull request:
This PR migrates stories for the `QuantitySelector` component and adds documentation for its props.

It also reverts a workaround introduced in #5354 in lieu of the official workaround recently published on the Storybook docs (see: https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#emotion11-quasi-compatibility).

Fixes #5367

### Testing

### Manual Testing

How to test the changes in this Pull Request:

1. Run `npm run storybook`
2. Explore the [QuantitySelector stories](http://localhost:6006/?path=/story/woocommerce-blocks-base-components-quantityselector--default). There should be two stories: a default one with the controlled component (i.e. can't set the values from the controls, but only from the action), a second one showing the component in its disabled state.

### User Facing Testing
These are steps for user testing (where "user" is someone interacting with this change that is not editing any code).
* [x] Same as above

sunyatasattva committed:
Add experimental Emotion11 compatibility

Revert the changes introduced in 087da0cf. The team over at Storybook
has recently implemented a workaround to enable Emotion11 compatibility
in 6.x releases.

See: https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#emotion11-quasi-compatibility

sunyatasattva committed:
Add `QuantitySelector` property docs and migrate stories

alexflorisca commented on assets/js/base/components/quantity-selector/stories/index.tsx:
```diff
@@ -0,0 +1,46 @@
+		itemName: 'widgets',
+	},
+	argTypes: {
+		quantity: { control: { disable: true } },
```
Why is the control for quantity disabled here? I suggest it might be useful to let a user vary the initial quantity with a control here

sunyatasattva replied:
@alexflorisca I've updated this, but because the way this component is supposed to be used is that the state is updated by its own plus/minus, though it still technically is a controlled component, it doesn't play in the best way with Storybook, and I had to use some magic to make it work.

Notice that, however, because of [this bug](https://github.com/storybookjs/storybook/issues/15924) you can change the number from the control and it will reflected in the component, but not viceversa.

alexflorisca commented on assets/js/base/components/quantity-selector/stories/index.tsx:
```diff
@@ -0,0 +1,46 @@
+	const [ quantity, setQuantity ] = useState( 1 );
+
+	const onChange = ( newVal: number ) => {
+		args.onChange?.( newVal );
```
I didn't know the optional chaining can be used as a check for a function here, that's cool!

alexflorisca reviewed (commented):
Nice work, just left a minor question/suggestion

sunyatasattva committed:
Expose the quantity control

*Note:* Because of a Storybook bug, the number input on the control panel of the
story, is not going to get updated by the component.

See: https://github.com/storybookjs/storybook/issues/15924

alexflorisca reviewed (approved):
Looks good 👍
